cloudant-sync-datastore-core/src/main/java/com/cloudant/common/RetriableTask.java

```diff
@@ -62,26 +60,27 @@ public int getTriesRemaining() {
 
     @Override
     public T call() throws Exception {
-        while (true) {
+        Exception lastException = null; // remember the exception we caught from the last try
+        do {
             try {
-                logger.fine("Retry #: " + triesRemaining + "," + task.toString());
+                logger.fine(String.format("%d tries remaining for task %s", triesRemaining, task));
                 T t = task.call();
-                if(triesRemaining < 3) {
-                    logger.fine("Success after retry: " + triesRemaining + ", " + task.toString());
-                }
+                logger.fine(String.format("Success with %d tries remaining for task %s",
+                        triesRemaining, task));
                 return t;
+            } catch (RetryException re) {
+                // if task.call() throws a RetryException this means we should exit early and not
+                // attempt any more retries - so just re-throw the exception
+                throw re;
             } catch (InterruptedException e) {
                 throw e;
             } catch (Exception e) {
-                triesRemaining--;
-                if (triesRemaining == 0) {
-                    throw new RetryException(totalRetries +
-                            " attempts to retry failed at " + timeToWait +
-                            "ms interval", e);
-                }
-                logger.log(Level.FINE,"Retry later: " + triesRemaining + ", " + task.toString(), e);
+                lastException = e;
                 Thread.sleep(timeToWait);
             }
-        }
+        } while (--triesRemaining > 0);
+        // if we got here, we ran out of retries
+        throw new RetryException(String.format("Failed after %d attempts for task %s",
+                totalRetries, task), lastException);
```

Comment:
@ricellis see af832a44f9977f313791206d9fe4803fbd7f9cae for this and the above comment. I've changed retries to tries in the appropriate places because that's what the tests imply. 

I also removed the "Success" message since I don't think it adds anything useful.